cqlsh_tests/test_cqlsh.py

```diff
@@ -127,6 +128,810 @@ def tearDown(self):
+        Test operations on a materialized view: create, describe, select from, drop, create using describe output.
+        @jira_ticket CASSANDRA-9961 and CASSANDRA-10348
+        """
+        self.cluster.set_configuration_options({'enable_materialized_views': 'true'})
+        self.cluster.populate(1)
+        self.cluster.start()
+        node1, = self.cluster.nodelist()
+        session = self.patient_cql_connection(node1)
+
+        create_ks(session, 'test', 1)
+
+        session.execute("""CREATE TABLE test.users (username varchar, password varchar, gender varchar,
+                    session_token varchar, state varchar, birth_year bigint, PRIMARY KEY (username))""")
+
+        session.execute("""CREATE MATERIALIZED VIEW test.users_by_state AS
+                    SELECT * FROM users WHERE STATE IS NOT NULL AND username IS NOT NULL PRIMARY KEY (state, username)""")
+
+        insert_stmt = "INSERT INTO users (username, password, gender, state, birth_year) VALUES "
+        session.execute(insert_stmt + "('user1', 'ch@ngem3a', 'f', 'TX', 1968);")
+        session.execute(insert_stmt + "('user2', 'ch@ngem3b', 'm', 'CA', 1971);")
+        session.execute(insert_stmt + "('user3', 'ch@ngem3c', 'f', 'FL', 1978);")
+        session.execute(insert_stmt + "('user4', 'ch@ngem3d', 'm', 'TX', 1974);")
+
+        describe_out, err = self.run_cqlsh(node1, 'DESCRIBE MATERIALIZED VIEW test.users_by_state')
+        assert 0 == len(err), err
+
+        select_out, err = self.run_cqlsh(node1, "SELECT * FROM test.users_by_state")
+        assert 0 == len(err), err
+        logger.debug(select_out)
+
+        drop_out, err = self.run_cqlsh(node1,
+                                       "DROP MATERIALIZED VIEW test.users_by_state; DESCRIBE KEYSPACE test; DESCRIBE table test.users")
+        assert 0 == len(err), err
+        assert "CREATE MATERIALIZED VIEW users_by_state" not in drop_out
+
+        describe_after_drop_out, err = self.run_cqlsh(node1, 'DESCRIBE MATERIALIZED VIEW test.users_by_state')
+        assert 0 == len(describe_after_drop_out.strip()), describe_after_drop_out
+        assert "Materialized view 'users_by_state' not found" in err
+
+        create_statement = 'USE test; ' + ' '.join(describe_out.splitlines()).strip()[:-1]
+        create_statement = self.strip_default_time_to_live(create_statement)
+        create_statement = self.strip_read_repair_chance(create_statement)
+        out, err = self.run_cqlsh(node1, create_statement)
+        assert 0 == len(err), err
+
+        reloaded_describe_out, err = self.run_cqlsh(node1, 'DESCRIBE MATERIALIZED VIEW test.users_by_state')
+        assert 0 == len(err), err
+        assert describe_out == reloaded_describe_out
+
+        reloaded_select_out, err = self.run_cqlsh(node1, "SELECT * FROM test.users_by_state")
+        assert 0 == len(err), err
+        assert select_out == reloaded_select_out
+
+
+@reuse_cluster
+class TestCqlshReuse(TestCqlshBase):
+    def try_reuse_node(self):
```

Comment:
Looks like we should unify on a single idiom as we have another pattern in `auth_test.py` with `clear_auth`.

Maybe something like `reset_cluster_between_tests` that becomes the idiomatic implementation we use? Or better yet, is called by default if you have the @reuse_cluster annotation specified? If we did that, we could define a default implementation that raises an exception in Tester so if someone tries to call it without remembering to extend that in their test class they get a hard failure / signal they're not using things correctly.